stratosphere/ncm/source/ncm_content_meta_database.hpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2019-2020 Adubbz, Atmosphère-NX
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include <switch.h>
+#include <stratosphere.hpp>
+
+namespace ams::ncm {
+
+    class ContentMetaDatabaseInterface : public IContentMetaDatabase {
+        public:
+            ContentMetaDatabaseInterface(ams::kvdb::MemoryKeyValueStore<ContentMetaKey>* kvs, const char* mount_name) : IContentMetaDatabase(kvs, mount_name) {
+            }
+            ContentMetaDatabaseInterface(ams::kvdb::MemoryKeyValueStore<ContentMetaKey>* kvs) : IContentMetaDatabase(kvs) {
+            }
+        private:
+            Result GetContentIdByTypeImpl(ContentId* out, const ContentMetaKey& key, ContentType type, std::optional<u8> id_offset);
+            Result GetLatestContentMetaKeyImpl(ContentMetaKey* out_key, ProgramId id);
+        public:
+            virtual Result Set(ContentMetaKey key, sf::InBuffer value) override;
```

Comment:
In this and other interfaces, literally everything taking in a ContentMetaKey should instead take in a const ContentMetaKey &.